pkgs/development/interpreters/python/wrapper.nix

```diff
@@ -8,7 +8,7 @@
 let
   recursivePthLoader = import ../../python-modules/recursive-pth-loader/default.nix { stdenv = stdenv; python = python; };
   env = let
-    paths = stdenv.lib.closePropagation (extraLibs ++ [ python recursivePthLoader ] ) ;
+    paths = extraLibs ++ [ python recursivePthLoader ];
```

Comment:
And how are the listed `extraLibs` Python packages now supposed to find their dependencies? This can't go in. 

I don't know if you intended to but changes like these require a separate pull request as this is a major change.

Reply:
Sure, this PR is just trying to discover how to get `ibus` fixed in mergeable fashion. `closePropagation` is deprecated anyway and should be replaced with something else. Could you explain why the environment has to have complete closure in it?

Reply:
We indeed need to get rid of the use of `closePropagation`. The issue we have is that we don't really mark packages providing Python libraries (read: they have `site-packages/`) as such, and we do not use such marking/attribute to determine dependencies. Therefore, a blunt approach is taken by just recursing through all `propagatedBuildInputs`. A possible solution would be https://github.com/NixOS/nixpkgs/pull/30606/files.

Due to how wrappers are created also more than needed remains inside closures.

Reply:
Well, I think we could filter out only those packages that have `site-packages/` at script runtime, and build environment with only those. That might require forking, or changing `buildEnv` Perl script, but doable. Is it a viable option?

Reply:
Not really. You have to track which Python packages belong together. E.g., imagine you have a new Python env that has in it a program that requires another Python env. You need to be able to distinguish those two. This is mostly an issue when you include in your env an application written in Python. There you may want to hide its `site-packages`.